### PR TITLE
Quick implementation of SSO login - Also handling of magic link

### LINKED ISSUE #64: Handle M_CONSENT_NOT_GIVEN error

### PR DESCRIPTION
**Pull request by bmarty**
- Refacto of LoginActivity to use ViewModel
- Handle SSO login
- Handle M_CONSENT_NOT_GIVEN error (Fixes #64)
- Handle magic link to configure homeserver url

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/login/LoginViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+        setState {
+            copy(
+                    asyncLoginAction = Success(Unit)
+            )
+        }
+    }
+
+    private fun handleSsoLoginSuccess(action: LoginActions.SsoLoginSuccess) {
+        val session = authenticator.createSessionFromSso(action.credentials, homeServerConnectionConfig!!)
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/riotx/features/login/LoginViewModel.kt]: !!

File: vector/src/main/java/im/vector/riotx/features/webview/WebViewEventListener.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 New Vector Ltd
```
Comment (BillCarsonFr) [inline vector/src/main/java/im/vector/riotx/features/webview/WebViewEventListener.kt]: 2019

File: vector/src/main/res/layout/activity_progress.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="?attr/vctr_waiting_background_color">
+
+    <ProgressBar
+        android:id="@+id/listView_spinner"
+        android:layout_width="40dp"
+        android:layout_height="40dp"
+        android:layout_gravity="center"
+        android:visibility="visible" />
```
Comment (BillCarsonFr) [inline vector/src/main/res/layout/activity_progress.xml]: ?

File: vector/src/main/java/im/vector/riotx/features/login/LoginViewModel.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/login/LoginViewModel.kt]: Done

File: vector/src/main/java/im/vector/riotx/features/webview/WebViewEventListener.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/webview/WebViewEventListener.kt]: Copied from Riot-Android

File: vector/src/main/res/layout/activity_progress.xml (hunk 3 above)
Comment (bmarty) [reply vector/src/main/res/layout/activity_progress.xml]: Cleaned up

**Pull request merged**